### PR TITLE
Remove looping and replace with dropna in validation.py

### PR DESCRIPTION
**Pull request by ktindiana**
Replaced lengthy looping approach with quick dropna approach.

### CODE REVIEW

**Comment by ktindiana** [comment]
I messed up with I made the commit. The message should be:

Improve removal of None values in validation.py

Replaced loops to remove None values in dataframes with dropna. 
This significantly reduces time to calculate metrics.

Fixed bug in match.py that lead to None values in best_eruption 
calculation which caused a crash with argmax. 
Cast null time difference to np.nan.

File: sphinxval/utils/match.py
Comment (rickyegeland) [inline sphinxval/utils/match.py:1]: These changes are off-topic.

File: sphinxval/utils/validation.py
Comment (rickyegeland) [inline sphinxval/utils/validation.py:1]: Looks much more efficient to me :-)

**Review by rickyegeland — COMMENTED** [review]
Interesting commit message :-) Will disappear with squash and merge, though.

Will approve pending successful test with Scoreboard dataset.

**Comment by rickyegeland** [comment]
Any commit that is +37 -292 is bound to be more efficient 😄

**Comment by rickyegeland** [comment]
Speedup on first test month of 202103 is 30x.  Approving this.

**Pull request merged**